contributed_definitions/nyaml/NXopt.yaml

```diff
@@ -427,152 +750,119 @@ NXopt(NXobject):
+          The information is recorded to qualify if the light used was likely
+          able to shine through the sample. 
+          
+          In this case the value should be set to the actual thickness of
+          the specimen viewed for an illumination situation where the nominal
+          surface normal of the specimen is parallel to the optical axis.
+        unit: NX_LENGTH
+      thickness_determination:
         exists: optional
         doc: |
-          Link to the NeXus file which describes the reference data if a
-          reference measurement was performed. Ideally, the reference
-          measurement was performed using the same conditions as the actual
-          measurement and should be as close in time to the actual measurement
-          as possible.
-      data_software(NXprocess):
+          If a thickness if given, please specify how this thickness was estimated or determined.
+      layer_structure:
         exists: optional
-        program:
-          doc: |
-            Commercial or otherwise defined given name of the program that was
-            used to generate the result file(s) with measured data and/or
-            metadata (in most cases, this is the same as INSTRUMENT/software).
-            If home written, one can provide the actual steps in the NOTE
-            subfield here.
-        version:
-          doc: |
-            Either version with build number, commit hash, or description of a
-            (online) repository where the source code of the program and build
-            instructions can be found so that the program can be configured in
-            such a way that result files can be created ideally in a
-            deterministic manner.
-        \@url:
-          exists: optional
-          doc: |
-            Website of the software.
-      (NXdata):
+        doc: |
+          Qualitative description of the layer structure for the sample,
+          starting with the top layer (i.e. the one on the front surface, on
+          which the light incident), e.g. native oxide/bulk substrate, or
+          Si/native oxide/thermal oxide/polymer/peptide.
+      sample_orientation:
         exists: optional
         doc: |
-          A plot of the multi-dimensional data array provided in
-          ENTRY/data/measured_data.
-        \@axes:
+          Specify the sample orientation, how is its sample normal oriented
+          relative in the laboratory reference frame, incident beam reference
+          frame.
+      substrate:
+        exists: recommended
+        doc: |
+          If the sample is grown or fixed on a substrate, specify this here by
+          a free text description.
+    (NXdata):
+      doc: |
+        A default view of the data provided in ENTRY/data_collection/measured_data. This
+        should be the part of the data set which provides the most suitable
+        representation of the data.
+
+        You may provide multiple instances of NXdata
+      \@axes:
```

Comment:
No need to specify the docs here (but can be kept for understanding the app-def).